### PR TITLE
Fix receiving property from Astarte

### PR DESCRIPTION
**Pull request by osmanhadzic**
Implement methods for publishing property to notify the caller.
Closes: #13

### CODE REVIEW

File: AstarteDeviceSDKCSharp/Protocol/AstarteServerPropertyInterface.cs [hunk 1]
```diff
@@ -18,18 +18,82 @@
...
+                        .InterfacePath(interfacePath)
+                        .Build();
+            }
+            else
+            {
+                astarteServerValue = null;
+            }
+
+            return astarteServerValue;
+        }
```
Comment (harlem88) [inline AstarteDeviceSDKCSharp/Protocol/AstarteServerPropertyInterface.cs:74]: I think you can move the `tagetMapping` block into the foreach and have a more compact block. 
```suggestion
    AstarteServerValue? astarteServerValue = null;

    foreach (var entry in GetMappings())
    {
        if (AstarteInterface.IsPathCompatibleWithMapping(interfacePath, entry.Key))
        {
            AstarteInterfaceMapping targetMapping = entry.Value;
            object? astarteValue = serverValue;
            if (targetMapping.GetType() == typeof(DateTime))
            {
               astarteValue = Convert.ToDateTime(serverValue);
            }

            astarteServerValue =
                    new AstarteServerValue.AstarteServerValueBuilder(astarteValue)
                   .InterfacePath(interfacePath)
                   .Build();
              break;           
        }
    }    
    return astarteServerValue;
 }
```

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: AstarteDeviceSDKCSharp/Protocol/AstarteServerPropertyInterface.cs (hunk 1 above)
Comment (osmanhadzic) [reply AstarteDeviceSDKCSharp/Protocol/AstarteServerPropertyInterface.cs]: I made changes.

File: AstarteDeviceSDKCSharp/Protocol/AstarteServerPropertyInterface.cs [hunk 2]
```diff
@@ -18,18 +18,71 @@
...
+        public AstarteServerValue? Build(string interfacePath, object? serverValue,
+        DateTime timestamp)
+        {
+            AstarteInterfaceMapping? targetMapping = null;
+            AstarteServerValue? astarteServerValue = null;
+            foreach (var entry in GetMappings())
+            {
+                if (AstarteInterface.IsPathCompatibleWithMapping(interfacePath, entry.Key))
+                {
+                    targetMapping = entry.Value;
```
Comment (harlem88) [inline AstarteDeviceSDKCSharp/Protocol/AstarteServerPropertyInterface.cs:49]: Move `AstarteInterfaceMapping` statement close to the use.
```suggestion
                    AstarteInterfaceMapping targetMapping = entry.Value;
```

**Review by harlem88 — CHANGES_REQUESTED** [review]

**Comment by harlem88** [comment]
Add A line below the Fixed section of changelog.md file:
```
- Fix receiving properties from Astarte.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [0.5.1] - Unreleased
 ### Fixed
 - Fix the construction of Pairing URL.
+- Fix the construction of Pairing URL.
```
Comment (harlem88) [inline CHANGELOG.md:10]: Remove this line is already present.
```suggestion
```

**Review by harlem88 — CHANGES_REQUESTED** [review]

**Review by harlem88 — APPROVED** [review]

**Review by nedimtokic — APPROVED** [review]

**Pull request merged**